operational_analysis/methods/plant_analysis.py

```diff
@@ -802,13 +846,50 @@ def filter_outliers(self, n):
             value_min=0.02 * plant_capac,
             value_max=1.2 * plant_capac,
         )
-
+        
+        if self.outlier_detection:
+            if self.time_resolution == 'M':
+                # Monthly linear regression (i.e., few data points): 
+                # filter outliers based on robust linear regression
+                # using Huber algorithm to flag outliers
+                X = sm.add_constant(df_sub[reanal])  # Reanalysis data with constant column
+                y = df_sub['gross_energy_gwh']*30/df_sub['num_days_expected']  # Energy data
+        
+                # Perform robust linear regression
+                rlm = sm.RLM(y, X, M=sm.robust.norms.HuberT(self._run.outlier_threshold))
+                rlm_results = rlm.fit()
+        
+                # Define valid data as points in which the Huber algorithm returned a value of 1
+                df_sub.loc[:, "flag_outliers"] = rlm_results.weights != 1
+            
+            else:
+                # Daily regressions (i.e., higher number of data points):
+                # Apply bin filter to catch outliers
+                df_sub.loc[:, "flag_outliers"] = filters.bin_filter(
+                        bin_col=df_sub["energy_gwh"],
+                        value_col=df_sub[reanal],
+                        bin_width=0.06 * plant_capac,
+                        threshold=int(round(self._run.outlier_threshold)),  # wind bin threshold (stdev outside the median)
```

Comment:
Why is the threshold converted to an int here? Based on the default values (2, 3.1), this would limit to thresholds of 2 or 3. But in `setup_monte_carlo_inputs`, you sample values in increments of 0.1. If these are only supposed to be ints, could you simplify the random number generation and mention this in the docstring for the `init` function?  Also, the comment here says stdev, but should it be something like "absolute deviation"?

Reply:
Thanks for catching this - int() was not necessary here and I have removed it, and I've changed the comment too.